src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,160 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> current = new Node<>(null, value, null);
+        if (size == 0) {
+            head = current;
+        } else {
+            tail.next = current;
+            current.prev = tail;
+        }
+        tail = current;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            if (size == 0) {
+                Node<T> current = new Node<>(null, value, null);
+                head = current;
+                tail = current;
+            } else {
+                Node<T> newHead = new Node<>(null, value, head);
+                newHead.next = head;
+                head.prev = newHead;
+                head = newHead;
+            }
+        } else if (index == size) {
+            Node<T> newTail = new Node<>(tail, value, null);
+            newTail.prev = tail;
+            tail.next = newTail;
+            tail = newTail;
+        } else {
+            Node<T> nodeByIndex = getNodeByIndex(index);
+            Node<T> middleCurrent = new Node<>(nodeByIndex.prev, value, nodeByIndex);
+            nodeByIndex.prev.next = middleCurrent;
+            middleCurrent.prev = nodeByIndex.prev;
+            nodeByIndex.prev = middleCurrent;
+            middleCurrent.next = nodeByIndex;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
```

Comment:
do not use abbreviated names
```suggestion
        for (T value : list) {
```